CORE-9318 Separate DB users for automatic bootstrap and runtime WIP

jacob-scott3141 opened the pull request.

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
added bootstrap db user

jacob-scott3141 committed:
updated e2e tests

jacob-scott3141 committed:
updated e2e tests

jacob-scott3141 committed:
updated e2e tests

jacob-scott3141 committed:
updated e2e tests

jacob-scott3141 committed:
updated e2e tests

josephzunigadaly committed:
CORE-10268: Fix liquibase error by implementing StreamResource.resolveSibling (#3151)

- Implement `resolveSibling`
- Add unit test
- Manually tested against QA Chat app

YashNabar committed:
CORE-9059 Manual approval of registration requests (#3099)

Introduces new REST endpoints for MGM for viewing registration requests, and approving/declining requests that are pending manual approval.

LWogan committed:
CORE-9342 Allow corda kafka consumers to reassemble chunks into compete records which were created by the corda kafka producer (#3040)

Update the CordaKafkaConsumer to handle chunked records which were generated by the CordaKafkaProducer.
Should ignore chunks if the consumer is expecting to read records of type Chunk. i.e sent via app level chunking.
Deserialization errors and checksum errors are handled within the ConsumerChunkDeserializerService which is built using the same onError handler as the deserializer.

Chunked records generated by the CordaKafkaProducer are always sent via transactions so ordering is always preserved with a single transactions. Therefore the final chunk always has the checksum and indicates when we can try to reassemble the chunk.

Chunks may be interleaved with chunks from a different transaction or normal records. We should not return any records from a partition which has partial chunks being read. Given chunks are always committed via transactions it should be impossible for a partition to be blocked by incomplete chunks. records are buffered in memory within the consumer per partition. This means if there is a rebalance the same records can be reread and no records will be lost. If our assignment changes after a rebalance we should remove any records buffered for partitions lost to ensure we do not have stale records if we are reassigned those partitions

lankydan committed:
CORE-10402 Make transaction builders mutable (#3155)

Confusion around the mutability of transaction builders was raised multiple
times which suggested we took the wrong approach of returning copies
each time.

We now mutate the existing instance and return it each time.

LWogan committed:
CORE-9516 flow session protocol chunking (#3147)

Updated the SessionData payload to allow bytes or chunks. When sending a session data message if it is too big then send it as chunks. We reuse the SessionEvent/SessionData envelope as the message needs to pass through various topics and across p2p. Using Session Events allows us to reuse the resend and dedup logic in the session protocol. We only reasemble chunks into the data events when the flow engine requests a chunk and all chunks are present.

P2P layer currently has no large avro message handling and has a hard limit of 5mb set via avro system property. Implementing handling at a session protocol layer allows the flow session messages to be delivered with smaller payloads

josephzunigadaly committed:
CORE-10216: Deduplicate the log messages from WorkerMonitorImpl (#3167)

Keep the previous message and only log new messages if they are different.

wzur-r3 commented on .ci/e2eTests/corda.yaml:
```diff
@@ -15,12 +24,14 @@ logging:
   cluster:
-    username:
-      value: "user"
+    host: "prereqs-postgresql"
```
why this change?

we use the same Helm chart to deploy up 3 different deployments, and fixing a name here is not a good idea

davidcurrie replied:
Probably my fault - I forget that there are a bunch of overrides being applied by the pipeline. The real problem here is that we now have duplication between the pipeline and `create.multi.clusters.sh`. The latter hasn't been kept up to date but we need to apply the same overrides in both places.

wzur-r3 commented on applications/workers/release/rpc-worker/src/e2eTest/resources/create.multi.clusters.sh:
```diff
@@ -13,17 +13,35 @@ create_cluster() {
+    oci://corda-os-docker.software.r3.com/helm-charts/corda-dev \
+    -f .ci/e2eTests/prereqs.yaml \
+    -f .ci/e2eTests/prereqs-eks.yaml \
+    --set 'kafka.replicaCount=1,kafka.zookeeper.replicaCount=1,kafka.offsetsTopicReplicationFactor=1,kafka.transactionStateLogReplicationFactor=1' \
```
please use one `--set` per value

wzur-r3 commented on applications/workers/release/rpc-worker/src/e2eTest/resources/install.aws.sh:
```diff
@@ -11,24 +11,34 @@ NAMESPACE="$USER-p2p-cluster"
+  oci://corda-os-docker.software.r3.com/helm-charts/corda-dev \
+  -f .ci/e2eTests/prereqs.yaml \
+  -f .ci/e2eTests/prereqs-eks.yaml \
+  --set 'kafka.replicaCount=1,kafka.zookeeper.replicaCount=1,kafka.offsetsTopicReplicationFactor=1,kafka.transactionStateLogReplicationFactor=1' \
```
please use one `--set` per value

wzur-r3 commented on values.yaml:
```diff
@@ -26,6 +26,16 @@ bootstrap:
+      password:
+        valueFrom:
+          secretKeyRef:
+            name: "prereqs-postgresql"
```
rather than hardcode the name of the secret here, please use variable in template as defaults, so multiple deployments in the same Kubernetes namespace don't interfere with each other

davidcurrie replied:
This one should be hard-coded. It's used by developers locally and is designed to work out of the box when they have provided no overrides to a prereqs install. There should be a similar change in `values-prereqs.yaml` though (for the new prereqs chart).

davidcurrie commented on .ci/e2eTests/corda.yaml:
```diff
@@ -15,12 +24,14 @@ logging:
     password:
       valueFrom:
         secretKeyRef:
+          name: "prereqs-postgresql"
```
The secret name here and also the one in `bootstrap.db` need to be overridden too as we have multiple postgres deployments.

josephzunigadaly committed:
CORE-8774: DB connection failures must be recoverable - Add scheduled DB check (#3081)

Add a reoccurring DB check that will notice the DB is down and bring components back up when it returns.

chrisr3 committed:
NOTICK: Configure Quasar for :libs:messaging in a single place. (#3176)

vlajos committed:
CORE-10226 Transaction verification exception cleanup (#3156)

CORE-10226 Rename TransactionVerificationException to TransactionSignatureException and remove it from Consensual verifications.

vlajos committed:
CORE-7237 Move TransactionMetadataVerifiers to ledger-...-data modules and use them in Ledger Transaction impls (#3172)

jacob-scott3141 committed:
changes to chart defaults

jacob-scott3141 committed:
changes to chart defaults

jacob-scott3141 committed:
changes to chart defaults

jacob-scott3141 committed:
changes to chart defaults

jacob-scott3141 committed:
changes to chart defaults

jacob-scott3141 committed:
added exporting of environment variables to multi cluster script

jacob-scott3141 committed:
added exporting of environment variables to multi cluster script

jacob-scott3141 committed:
added exporting of environment variables to multi cluster script

jacob-scott3141 committed:
added updated variable names in charts

jacob-scott3141 committed:
added updated variable names in charts

jacob-scott3141 committed:
added updated variable names in charts

jacob-scott3141 committed:
added updated variable names in charts

jacob-scott3141 committed:
added updated variable names in charts

jacob-scott3141 committed:
added updated variable names in charts

josephzunigadaly committed:
CORE-8774: Turn off DB check while we investigate a failing test (#3179)

davidcurrie commented on applications/workers/release/rpc-worker/src/e2eTest/resources/create-clusters.sh:
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
```
where did this script come from? doesn't it do the same thing as `create.multi.clusters.sh` but more verbosely?

davidcurrie commented on .ci/e2eTests/corda.yaml:
```diff
@@ -1,4 +1,12 @@
+  db:
+    cluster:
+      username:
+        value: "postgres"
```
this will fail in the CI until `bootstrap.db.cluster.password.valueFrom.secretKeyRef.name` is set in the shared pipeline script

jacob-scott3141 committed:
Remove temporary testing script

jacob-scott3141 committed:
updated pipelines and fixed bootstrap password override in multi cluster script

jacob-scott3141 committed:
updated pipelines and fixed bootstrap password override in multi cluster script

jacob-scott3141 committed:
updated pipelines and fixed bootstrap password override in multi cluster script

jacob-scott3141 committed:
Merge branch 'jacob/CORE-9318-v2' of https://github.com/corda/corda-runtime-os into jacob/CORE-9318-v2

jacob-scott3141 committed:
added permission to user DB_CLUSTER_USERNAME

jacob-scott3141 committed:
added permission to user DB_CLUSTER_USERNAME

jacob-scott3141 committed:
added permission to user DB_CLUSTER_USERNAME

yarivus commented on .ci/e2eTests/Jenkinsfile:
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.0') _
+@Library('corda-shared-build-pipeline-steps@jacob/CORE-9318') _
```
this should be temp

yarivus reviewed (commented):
See pending comments

jacob-scott3141 committed:
Merge branch 'jacob/CORE-9318-v2' of https://github.com/corda/corda-runtime-os into jacob/CORE-9318-v2

jacob-scott3141 committed:
Merge branch 'jacob/CORE-9318-v2' of https://github.com/corda/corda-runtime-os into jacob/CORE-9318-v2
